app/controllers/users_controller.rb

```diff
@@ -5,9 +5,18 @@ class UsersController < ApplicationController
   before_action :admin_user,     only: :destroy
 
   def index
-    @users = User.paginate(page: params[:page])
+    if params[:name] != nil
+      @users = User.where(name: params[:name])
+      if @users.size == 0
+        @users = nil
+      else
+        @users = User.where(name: params[:name]).paginate(page: params[:page])
+      end
+    else
+      @users = User.paginate(page: params[:page])
+    end
```

Comment:
インターンシップお疲れ様です、NaClの野口です。

こちらのPRを見かけて気になったところがあったのでコメントします。
（あくまで書き方の話なので、修正を取り込むかどうかの判断はお任せします:pray:

---

条件を整理すると、以下のように書けそうです。
**補足:** 検索条件が空の場合、全件表示するシステムが多いです。

```suggestion
    @users = User.paginate(page: params[:page])
    @users.where!(name: params[:name]) if params[:name].present?
```

上記修正に含まれる `present?` はRailsが定義しているメソッドなのですが、条件を書くときに頻出するので覚えておいて損はないです。

参考情報: [【Rails入門】presentメソッドの使い方まとめ | 侍エンジニアブログ](https://www.sejuku.net/blog/27232)